### DISCUSSION TITLE
Are there API references for `fields` and  `filters` packages?

### DISCUSSION THREAD

**Discussion by devraj** [betodealmeida/shillelagh#450] [Q&A]
First of all thank you @betodealmeida for all your work on this. We're integrating this in our project for a narrow use case (https://github.com/anomaly/gallagher/issues/31). While the `fields` and `filters` packages are very well documented inline.

I was wondering if there are any API references published as part of the documentation on the web? Apologies in advance if I missed something in the current documentation.

> My use case is writing a custom adapter which will integrate heavily into a pydantic/httpx based API client that talks to the service.

**Comment by betodealmeida** [comment]
Hi, @devraj! Sorry for the delay, I thought I would get pinged on this on my email but I didn't.

There are no docs for those packages, the only reference for now is the code itself. But the `Field` class has a pretty long docstring explaining why it's needed and how it works, as well as extensive comments. The filters are a Python representation of the SQLite constraint:

```python
# map between APSW operators and the ``Operator`` enum
operator_map = {
    apsw.SQLITE_INDEX_CONSTRAINT_EQ: Operator.EQ,
    apsw.SQLITE_INDEX_CONSTRAINT_GE: Operator.GE,
    apsw.SQLITE_INDEX_CONSTRAINT_GT: Operator.GT,
    apsw.SQLITE_INDEX_CONSTRAINT_LE: Operator.LE,
    apsw.SQLITE_INDEX_CONSTRAINT_LT: Operator.LT,
}
```

And then:

```
class Equal(Filter):
    """
    Equality comparison.
    """

    operators: Set[Operator] = {
        Operator.EQ,
    }

    def __init__(self, value: Any):
        self.value = value
```

So that when SQLite finds `foo = 42` it will pass that to the adapter as `Equal(42)` in the `get_data()` method.
